pytorch_lightning/strategies/single_device.py

```diff
@@ -88,6 +90,14 @@ def barrier(self, *args, **kwargs) -> None:
     def broadcast(self, obj: object, src: int = 0) -> object:
         return obj
 
+    @classmethod
+    def register_strategies(cls, strategy_registry: dict) -> None:
+        strategy_registry.register(
+            cls.distributed_backend,
+            cls,
+            description=f"{cls.__class__.__name__} Strategy",
+        )
+
```

Comment:
I would go with the second suggestion. `strategy="single_device"` would be unnecessary, as it is implied by `devices=1`

Reply:
but would it be wrong?

Couldn't you theoretically also spin up a DDP-Setting with just one process?

I agree that this should mean the same, but I think since this is not wrong, we should support it.

Reply:
Agree with @justusschock, all the existing strategy should have a name, and if user choose to pass in, we should support it.
And user can pass `strategy=SingleDeviceStrategy()` even we don't define a str for it. 

Same as mentioned in ipu.py L65, strategies registry behavior should be consistent.
For single device, user can pass in strategy=